### ISSUE TITLE
Use conda lock file for deployment env

### ISSUE THREAD

**Issue by mikemhenry** [OpenFreeEnergy/alchemiscale#51]
See https://github.com/mamba-org/micromamba-docker#using-a-lockfile

Low priority, #25 and #49 should happen first

**Comment by dotsdl** [comment]
I think we can cut one of these alongside releases. Perhaps derive from `docker.yml` and call it `docker-prod.yml`?
